Sem. 10/01/Car/Car.cpp

```diff
@@ -0,0 +1,181 @@
+#include "Car.h"
+
+static const short MAX_TYRES_COUNT = 4;
+
+Car::Car():
+	fuelTank(),
+	engine(),
+	battery(),
+	tyres(),
+	kmDriven(0.0),
+	weight(0.0)
+{
+}
+
+Car::~Car()
+{
+	free();
+}
+
+Car::Car(const Car& other)
+{
+	copyFrom(other);
+}
+
+Car& Car::operator=(const Car& other)
+{
+	if (this != &other)
+	{
+		free();
+		copyFrom(other);
+	}
+	return *this;
+}
+
+Car::Car(Car&& other) noexcept
+{
+	moveFrom(std::move(other));
+}
+
+Car& Car::operator=(Car&& other) noexcept
+{
+	if (this != &other)
+	{
+		free();
+		moveFrom(std::move(other));
+	}
+	return *this;
+}
+
+Car::Car(const FuelTank& fuelTank, const Engine& engine, 
+	Tyre* tyres, const AutomativeBattery& battery, 
+	double km, double weight)
+	: fuelTank(fuelTank), engine(engine), tyres(tyres), battery(battery),
+	kmDriven(km), weight(weight) {}
```

Comment:
This way you are copying the Tyre * which leads to 2 pointers pointing to 1 memory block